From 8c603f4d42b89bb8b2b0d7554e536ba35d0148f3 Mon Sep 17 00:00:00 2001
From: Zongdong Jiao <zongdong.jiao@amlogic.com>
Date: Tue, 17 Dec 2013 10:36:28 +0800
Subject: [PATCH 2649/5965] PD #84102: hdmitx: clk: re-write hdmi phy reset
 code under vlsi

PD #84102: hdmitx: clk: reset hdmi phy three times
---
 drivers/amlogic/display/vout/enc_clk_config.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/drivers/amlogic/display/vout/enc_clk_config.c b/drivers/amlogic/display/vout/enc_clk_config.c
index a4a44698908a..80d602368963 100755
--- a/drivers/amlogic/display/vout/enc_clk_config.c
+++ b/drivers/amlogic/display/vout/enc_clk_config.c
@@ -94,12 +94,16 @@ static void set_hpll_clk_out(unsigned clk)
             break;
     }
 #ifdef CONFIG_ARCH_MESON8
-    aml_write_reg32(P_HHI_HDMI_PHY_CNTL1, 0);
-    aml_write_reg32(P_HHI_HDMI_PHY_CNTL1, 1);       // Soft Reset HDMI PHY
-    h_delay();
-    aml_write_reg32(P_HHI_HDMI_PHY_CNTL1, 0);
-    h_delay();
-    aml_write_reg32(P_HHI_HDMI_PHY_CNTL1, 2);       // Enable HDMI PHY
+    // P_HHI_HDMI_PHY_CNTL1     bit[1]: enable clock    bit[0]: soft reset
+#define RESET_HDMI_PHY()                        \
+    aml_write_reg32(P_HHI_HDMI_PHY_CNTL1, 3);   \
+    h_delay();                                  \
+    aml_write_reg32(P_HHI_HDMI_PHY_CNTL1, 2);   \
+    h_delay()
+
+    RESET_HDMI_PHY();
+    RESET_HDMI_PHY();
+    RESET_HDMI_PHY();
 #endif
 }
 
-- 
2.19.0

